vault/logical_system_activity.go

```diff
@@ -288,6 +300,17 @@ func (b *SystemBackend) handleActivityConfigUpdate(ctx context.Context, req *log
 		return nil, err
 	}
 
+	// Handle attempts to update read-only fields
+	{
+		if _, ok := req.Data["reporting_enabled"]; ok {
```

Comment:
Is "reporting_enabled" only set upon start up using the HCL configuration?
I am wondering if we could ignore caring about these two since they should not be updated using this API? Like why do we even include them in the API parameters if we are going to say, hey you cannot do that?

Reply:
This is a way for us to expose the internal configs for the UI -- this endpoint is currently in use for the data usage page. I thought about ignoring the `UPDATE`, but after some discussion in Slack, it seemed like the preferred approach is to prevent modification to these fields explicitly.

So the idea here is these fields are exposed via Read, but if you attempt to Update any of the fields, you'll get these errors.

Reply:
I see. Who is going to consume the explicit errors though for the update case? Like what is the use case here?